### PR TITLE
Refs #3023: Adds plugin rake task to extract translations.

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by domcleal** [comment]
@lzap could you review please?  It might better to ref #2942 or just open a new ticket.

File: lib/foreman/gettext/support.rb [hunk 1]
```diff
@@ -1,58 +1,63 @@
...
-  def self.detect_locale_type
-    if Rails.env.development?
-      :po
-    else
-      :mo
-    end
-  end
+module Foreman
+  module Gettext
+    module Support
```
Comment (lzap) [inline lib/foreman/gettext/support.rb:5]: Thanks, I will test this against Discovery. Btw does this change make any difference? Just interested.

File: lib/foreman/gettext/support.rb [hunk 2]
```diff
@@ -1,58 +1,63 @@
...
-      # TRANSLATORS: Provide locale name in native language (e.g. English, Deutsch or Português)
-      human_locale = _("locale_name")
-      human_locale = locale if human_locale == "locale_name"
-      FastGettext.human_available_locales << [ human_locale, locale ]
     end
-  ensure
-    FastGettext.locale = original_locale
   end
-
 end
```
Comment (lzap) [inline lib/foreman/gettext/support.rb:63]: Hmm the whole patch is rather unreadable, why you changed the indentation?

Comment (lzap) [inline lib/foreman/gettext/support.rb:63]: Got it :)

**Comment by lzap** [comment]
Works out of the box! Excellent.

**Comment by lzap** [comment]
Merged as 432db23 for Foreman 1.5, thanks @ehelms !

**Pull request closed without merging**